components/pd_client/src/util.rs

```diff
@@ -220,17 +223,18 @@ impl Client {
     ///
     /// If `force` is false, it will reconnect only when members change.
     pub async fn reconnect(&self, force: bool) -> Result<()> {
-        let (future, start) = {
+        let start = Instant::now();
+
+        let future = {
             let inner = self.inner.rl();
-            if inner.last_update.elapsed() < Duration::from_secs(RECONNECT_INTERVAL_SEC) {
+            if start.duration_since(inner.last_try_reconnect) < GLOBAL_RECONNECT_INTERVAL {
                 // Avoid unnecessary updating.
-                return Ok(());
+                // Prevent a large number of reconnections in a short time.
+                return Err(box_err!("cancel reconnection due to too small interval"));
```

Comment:
Why returning `Error` instead of `Ok` just like the past?

Reply:
If the request is received `reconnect ok`, it will try to resend the request immediately and will not `sleep 1s`. At present, both `async request` and `sync request` have been changed to send the request first and then try to reconnect.